src/TestFramework/Factory.php

```diff
@@ -124,21 +143,43 @@ public function create(string $adapterName, bool $skipCoverage): AbstractTestFra
                     $this->infectionConfig->getSourceDirs(),
                     $skipCoverage
                 ),
-                new MutationConfigBuilder($this->tmpDir, $phpUnitConfigContent, $this->xmlConfigurationHelper, $this->projectDir),
+                new MutationConfigBuilder($this->tmpDir, $phpUnitConfigContent, $this->xmlConfigurationHelper, $this->projectDir, new JUnitTestCaseSorter()),
                 new ArgumentsAndOptionsBuilder(),
-                $this->versionParser
+                $this->versionParser,
+                $this->commandLineBuilder
             );
         }
 
         if ($adapterName === TestFrameworkTypes::PHPSPEC) {
             $phpSpecConfigPath = $this->configLocator->locate(TestFrameworkTypes::PHPSPEC);
 
             return new PhpSpecAdapter(
-                new TestFrameworkFinder(TestFrameworkTypes::PHPSPEC),
+                (new TestFrameworkFinder(TestFrameworkTypes::PHPSPEC))->find(),
                 new PhpSpecInitialConfigBuilder($this->tmpDir, $phpSpecConfigPath, $skipCoverage),
                 new PhpSpecMutationConfigBuilder($this->tmpDir, $phpSpecConfigPath, $this->projectDir),
                 new PhpSpecArgumentsAndOptionsBuilder(),
-                $this->versionParser
+                $this->versionParser,
+                $this->commandLineBuilder
+            );
+        }
+
+        if ($adapterName === TestFrameworkTypes::CODECEPTION) {
+            $codeceptionConfigPath = $this->configLocator->locate(TestFrameworkTypes::CODECEPTION);
+            $codeceptionConfigContent = file_get_contents($codeceptionConfigPath);
+            $codeceptionConfigContentParsed = Yaml::parse($codeceptionConfigContent);
```

Comment:
Hm what if the YAML content is invalid? Do we have a graceful/understandable failure there?

Reply:
added a check, now it will say the following:

<img width="739" alt="infection_yaml_parse" src="https://user-images.githubusercontent.com/3725595/69353552-bb45ec80-0c8f-11ea-9f59-85385fcee77a.png">